File: TotalAdmin/TotalAdmin.Model/Entities/PurchaseOrder.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TotalAdmin.Model.Entities
{
    public class PurchaseOrder : BaseEntity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int PoNumber { get; set; }

        [Display(Name = "Creation Date")]
        [Required(ErrorMessage = "Please enter a Creation date.")]
        [DataType(DataType.Date)]
        [DisplayFormat(DataFormatString = "{0: yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime CreationDate { get; set; }

        [ConcurrencyCheck]
        public byte[]? RowVersion { get; set; }

        [Display(Name = "Employee Number")]
        [Required(ErrorMessage = "Please select a employee.")]
        public int EmployeeNumber { get; set; }
        public string EmployeeName { get; set; }
        public string EmployeeSupervisorName { get; set; }
        public string EmpDepartmentName { get; set; }
        public string PurchaseOrderStatus { get; set; }

        [Display(Name = "Status")]
        [Required(ErrorMessage = "Please select a status.")]
        public int StatusId { get; set; }

        // Navigation property
        public ICollection<Item>? Items { get; set; }
        public bool HasMergeOccurred { get; set; }
        public string FormattedPoNumber { get; set; }
        public string EmployeeEmail { get; set; }

        // Derived properties
        public decimal Subtotal => Items?.Sum(item => item.Subtotal) ?? 0;
        public decimal Tax => Items?.Sum(item => item.Tax) ?? 0;
        public decimal GrandTotal => Items?.Sum(item => item.GrandTotal) ?? 0;
    }
}

File: AndroidFrontEnd/totaladmin/lib/models/purchaseOrder.dart
import 'item.dart';

class PurchaseOrder {
  final int poNumber;
  final DateTime creationDate;
  final int employeeNumber;
  final String? employeeName;
  final String? employeeSupervisorName;
  final String? empDepartmentName;
  final String? purchaseOrderStatus;
  final int statusId;
  final List<Item>? items;
  final bool? hasMergeOccurred;
  final String? formattedPoNumber;
  final double? grandTotal;
  final double? subtotal;
  final double? tax;
  final double? totalExpenseAmt;
  final List<dynamic>? errors;
  final String? employeeEmail;

  PurchaseOrder({
    required this.poNumber,
    required this.creationDate,
    required this.employeeNumber,
    this.employeeName,
    this.employeeSupervisorName,
    this.empDepartmentName,
    this.purchaseOrderStatus,
    required this.statusId,
    this.items,
    this.hasMergeOccurred,
    this.formattedPoNumber,
    this.grandTotal,
    this.subtotal,
    this.tax,
    this.totalExpenseAmt,
    this.errors,
    this.employeeEmail,
  });

  factory PurchaseOrder.fromJson(Map<String, dynamic> json) {
    return PurchaseOrder(
      poNumber: json['poNumber'],
      creationDate: DateTime.parse(json['creationDate']),
      employeeNumber: json['employeeNumber'],
      employeeName: json['employeeName'] as String?,
      employeeSupervisorName: json['employeeSupervisorName'] as String?,
      empDepartmentName: json['empDepartmentName'] as String?,
      purchaseOrderStatus: json['purchaseOrderStatus'] as String?,
      statusId: json['statusId'],
      items: (json['items'] as List<dynamic>?)
          ?.map((item) => Item.fromJson(item))
          .toList(),
      hasMergeOccurred: json['hasMergeOccurred'] as bool?,
      formattedPoNumber: json['formattedPoNumber'] as String?,
      grandTotal: (json['grandTotal'] as num?)?.toDouble(),
      subtotal: (json['subtotal'] as num?)?.toDouble(),
      tax: (json['tax'] as num?)?.toDouble(),
      totalExpenseAmt: (json['totalExpenseAmt'] as num?)?.toDouble(),
      errors: json['errors'] as List<dynamic>?,
      employeeEmail: json['employeeEmail'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'poNumber': poNumber,
      'creationDate': creationDate.toIso8601String(),
      'employeeNumber': employeeNumber,
      'employeeName': employeeName,
      'employeeSupervisorName': employeeSupervisorName,
      'empDepartmentName': empDepartmentName,
      'purchaseOrderStatus': purchaseOrderStatus,
      'statusId': statusId,
      'items': items?.map((item) => item.toJson()).toList(),
      'hasMergeOccurred': hasMergeOccurred,
      'formattedPoNumber': formattedPoNumber,
      'grandTotal': grandTotal,
      'subtotal': subtotal,
      'tax': tax,
      'totalExpenseAmt': totalExpenseAmt,
      'errors': errors,
      'employeeEmail': employeeEmail,
    };
  }

  @override
  String toString() {
    return 'PurchaseOrder{poNumber: $poNumber, creationDate: $creationDate, employeeNumber: $employeeNumber, employeeName: $employeeName, employeeSupervisorName: $employeeSupervisorName, empDepartmentName: $empDepartmentName, purchaseOrderStatus: $purchaseOrderStatus, statusId: $statusId, items: $items, hasMergeOccurred: $hasMergeOccurred, formattedPoNumber: $formattedPoNumber, grandTotal: $grandTotal, subtotal: $subtotal, tax: $tax, totalExpenseAmt: $totalExpenseAmt, errors: $errors, employeeEmail: $employeeEmail}';
  }
}
